common/azure-common-test/src/main/java/com/azure/common/TestBase.java

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.common;
+
+import com.azure.common.utils.SdkContext;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.rules.TestName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Base class for running live and playback tests using {@link InterceptorManager}.
+ */
+public abstract class TestBase {
```

Comment:
Should we also add placeholders for BeforeClass and AfterClass? These could allow performance wins with the tests instead of having to instantiate common build for every test.

Reply:
I thought about it, but the existing TestBase had a BeforeClass and AfterClass that didn't add value. I'm not sure how performance could be improved with having the base class with a no-op `BeforeClass` that then calls the implementing class.

Reply:
That was left a little too vague on my part, less on the stub methods being useful now but more on they would allow optimizations. For example the App Config unit tests create a new builder, run an authenticate credentials, and construct a new client for each test. That could just get done once during the beforeClass call.

Reply:
Ahh I see what you're saying. Looking into the BeforeClass annotation... unfortunately it has to be a static method. So, it doesn't lend itself to having a stub method that can be overridden in an implementing class. :/